crates/common/src/auction/README.md

```diff
@@ -0,0 +1,233 @@
+# Auction Orchestration System
+
+A flexible, extensible framework for managing multi-provider header bidding auctions with support for parallel execution and mediation.
+
+## Overview
+
+The auction orchestration system allows you to:
+- Run multiple auction providers (Prebid, Amazon APS, Google GAM, etc.) in parallel or sequentially
+- Implement mediation strategies where a primary ad server (like GAM) makes the final decision
+- Configure different auction flows for different scenarios
+- Easily add new auction providers
+
+## Architecture
+
+```
+┌─────────────────────────────────────────────────────────┐
+│                  Auction Orchestrator                   │
+│  - Manages auction workflow & sequencing                │
+│  - Combines bids from multiple sources                  │
+│  - Applies business logic                               │
+└─────────────────────────────────────────────────────────┘
+                          │
+                          │ uses
+                          ▼
+┌─────────────────────────────────────────────────────────┐
+│              AuctionProvider Trait                       │
+│  - request_bids()                                        │
+│  - provider_name()                                       │
+│  - timeout_ms()                                          │
+│  - is_enabled()                                          │
+└─────────────────────────────────────────────────────────┘
+                          │
+        ┌─────────────────┼─────────────────┐
+        │                 │                 │
+        ▼                 ▼                 ▼
+  ┌──────────┐      ┌──────────┐     ┌──────────┐
+  │  Prebid  │      │ Amazon   │     │  Google  │
+  │ Provider │      │   APS    │     │   GAM    │
+  └──────────┘      └──────────┘     └──────────┘
+```
+
+## Key Concepts
+
+### Auction Provider
+Implements the `AuctionProvider` trait to integrate with a specific SSP/ad exchange.
+
+### Auction Flow
+A named configuration that defines:
+- Which providers participate
+- Execution strategy (parallel, waterfall, etc.)
+- Timeout settings
+- Optional mediator
+
+### Orchestrator
+Manages the execution of an auction flow, coordinates providers, and collects results.
+
+## Auction Strategies
+
+### 1. Parallel + Mediation (Recommended)
+**Use case:** Header bidding with GAM as primary ad server
+
+```toml
+[auction]
+enabled = true
+strategy = "parallel_mediation"
+bidders = ["prebid", "aps"]
+mediator = "gam"
+timeout_ms = 2000
+```
+
+**Flow:**
+1. Prebid and APS run in parallel
+2. Both return their bids simultaneously
+3. Bids are sent to GAM for final mediation
+4. GAM competes its own inventory and returns winning creative
+
+### 2. Parallel Only
+**Use case:** Client-side auction, no mediation
+
+```toml
+[auction]
+enabled = true
+strategy = "parallel_only"
+bidders = ["prebid", "aps"]
+timeout_ms = 2000
+```
+
+**Flow:**
+1. All bidders run in parallel
+2. Highest bid wins
```

Comment:
This may be addressed elsewhere, but AFAIK APS doesn't return a clear text creative bid, so we'll need to understand what the decryption of the TAM value looks like